### DISCUSSION TITLE
What will aioble scanResult.name() report for an advertisement containing unprintable characters?

### DISCUSSION THREAD

**Discussion by brianreinhold** [micropython/micropython#13750] [Libraries & Drivers] [answered]
I have  device that is sending the following advertisment
```
0f 09 31 53 43 31 30 30 00 00 00 00 00 00 00 00 03 19 00 00 02 01 06 03 03 02 78
```
Note the device name
```
0f  // length of data including code
09 // adv code indicating device name
31 53 43 31 30 30 00 00 00 00 00 00 00 00 
```
The name has a whole slew of unprintable null characters at the end. What will ScanResult.name() return? Is it
`1SC100` or `1SC100        ` (replace the nulls with spaces 0x20)?

**Comment by bixb922** [comment, marked as answer]
ScanResult.name() will interpret (decode) the name as UTF-8 string using str(name,'utf8'). The trailing nulls will remain as null characters at the end of the string. To be sure to have the "ascii only" characters, replace the nulls by "":
```python
>>> advertised_name=b'1SC100\x00\x00\x00\x00\x00\x00\x00\x00'
>>> name_as_string=str(advertised_name,"utf-8")
>>> name_as_string
'1SC100\x00\x00\x00\x00\x00\x00\x00\x00'
>>> trimmed_name=name_as_string.replace("\x00","")
>>> trimmed_name
'1SC100'
```

**Reply by brianreinhold** [reply]
@bixb922 I ended up doing that - actually I used 
`trimmed_name=name_as_string.replace('\0', '')`
but it seemed to work. Maybe both are valid?

**Reply by bixb922** [reply]
Perfect, both are valid!
